intelmq/lib/bot.py

```diff
@@ -952,6 +954,25 @@ def __init__(self, bot_id: str, start: bool = False, sighup_event=None,
             self.stop()
         self.group = 'Parser'
 
+        # validate default fields
+        stop = False
+        if self.default_fields:
+            dummy_event = self.new_event()
+            for key, value in self.default_fields.items():
+                try:
+                    dummy_event.add(key, value, raise_failure=True)
+
+                except exceptions.InvalidValue:
+                    self.logger.error("Invalid value of key '%s' in default_fields parameter.", key)
+                    stop = True
+
+                except exceptions.InvalidKey:
+                    self.logger.error("Invalid key '%s' in default_fields parameter.", key)
+                    stop = True
+
+            if stop:
+                self.stop()
```

Comment:
do we really want to stop a bot if an invalid key is given? What about logging it as warning and ignoring it?
I am just a bit extra careful since this is in the main Bot class.

Reply:
I believe so. This check uses the same code that is used for the actual parsing: `event.add_key`. If an invalid key is added during the actual parsing then the bot would crash, stop and restart on every event anyway.

Also when the bots starts a and runs without crashing/stopping (but ignoring invalid key) I wouldn't check the logs assuming everything is fine.

Reply:
okay, since it's configured at start time via config file, at least a stop() will be noticed immediately. So, from the UX side, it's clear that something does not work. Yeah, okay... convinced. Thanks for thinking this through and for the PR! Merging...